packages/@webex/plugin-cc/src/services/task/index.ts

```diff
@@ -0,0 +1,83 @@
+import {LocalMicrophoneStream} from '@webex/calling';
+import {getErrorDetails} from '../core/Utils';
+import {LoginOption} from '../../types';
+import {CC_FILE} from '../../constants';
+import routingContact from './contact';
+import {ITask, TaskId, TaskResponse, TaskData} from './types';
+import WebCallingService from '../WebCallingService';
+
+export default class Task implements ITask {
+  private contact: ReturnType<typeof routingContact>;
+  private localAudioStream: LocalMicrophoneStream;
+  private webCallingService: WebCallingService;
+  public data: TaskData;
+
+  constructor(
+    contact: ReturnType<typeof routingContact>,
+    webCallingService: WebCallingService,
+    data: TaskData
+  ) {
+    this.contact = contact;
+    this.data = data;
+    this.webCallingService = webCallingService;
+  }
+
+  /**
+   * This is used for incoming task accept by agent.
+   *
+   * @param taskId - Unique Id to identify each task
+   *
+   * @returns Promise<TaskResponse>
+   * @throws Error
+   * @example
+   * ```typescript
+   * task.accept(taskId).then(()=>{}).catch(()=>{})
+   * ```
+   */
+  public async accept(taskId: TaskId): Promise<TaskResponse> {
+    try {
+      if (this.webCallingService.loginOption === LoginOption.BROWSER) {
+        const constraints = {
+          audio: true,
+        };
+
+        const localStream = await navigator.mediaDevices.getUserMedia(constraints);
+        const audioTrack = localStream.getAudioTracks()[0];
+        this.localAudioStream = new LocalMicrophoneStream(new MediaStream([audioTrack]));
+        this.webCallingService.answerCall(this.localAudioStream, taskId);
+
+        return Promise.resolve(); // TODO: Update this with sending the task object received in AgentContactAssigned
+      }
+
+      // TODO: Invoke the accept API from services layer. This is going to be used in Outbound Dialer scenario
+      return this.contact.accept({interactionId: taskId});
```

Comment:
Should we be changing the state of the agent when the call is accepted?

Reply:
@Kesari3008 have you tested this? Is there an event that arrives when we answer the call from WxCC backend?

Reply:
Resolved through testing. Agent is automatically marked busy on the backend.